src/view/gui/fMainWindow.cpp

```diff
@@ -1823,10 +1823,28 @@ void fMainWindow::LoadSlicerImages(const std::string &fileName, const int &image
     {
       image4DSlider->setEnabled(true);
       image4DSlider->setRange(0, imageInfo.GetImageSize()[3] - 1);
+      image4DMaxSliceIndicator->setText(QString::fromStdString(" / " + std::to_string(imageInfo.GetImageSize()[3])));
       ImageTypeFloat4D::Pointer imagePerf = cbica::ReadImage<ImageTypeFloat4D>(fname);
-      imageManager->SetPerfImage(imagePerf);
+      
+      
       imageManager->mImageSubType = CAPTK::ImageModalityType::IMAGE_TYPE_PERFUSION;
-      //return;
+      imageManager->SetOriginalOrigin(imageInfo.GetImageOrigins()); // Fix missing (3D) origins for 4D
+      /*auto originalDirection = imageInfo.GetImageDirections();
+      ImageTypeFloat3D::DirectionType originalDirectionIn3D;
+      for (int i = 0; i < 3; i++)
+      {
+          for (int j = 0; j < 3; j++)
+          {
+              originalDirectionIn3D[i][j] = originalDirection[i][j];
+          }
+      }
+      imageManager->SetOriginalDirection(originalDirectionIn3D);
+      */
+      //auto tempImage = cbica::GetImageOrientation< ImageTypeFloat4D >(imagePerf);
+      //imageManager->SetOriginalOrientation(tempImage.first);
+      //imagePerf = ChangeImageDirectionToIdentity< ImageTypeFloat4D >(tempImage.second);
```

Comment:
Any reason for this commented code to be present? If yes, then please place a comment here.